biz.aQute.bndlib/src/aQute/bnd/osgi/Verifier.java

```diff
@@ -673,7 +675,12 @@ public void verify() throws Exception {
 		verifyDirectives(Constants.IMPORT_PACKAGE, "resolution:", PACKAGEPATTERN, "package");
 		verifyDirectives(Constants.REQUIRE_BUNDLE, "visibility:|resolution:", SYMBOLICNAME, "bsn");
 		verifyDirectives(Constants.FRAGMENT_HOST, "extension:", SYMBOLICNAME, "bsn");
-		verifyDirectives(Constants.PROVIDE_CAPABILITY, "effective:|uses:", null, null);
+		verifyDirectives(Constants.PROVIDE_CAPABILITY, "effective:|uses:", null, null, (namespace, pattern) -> {
```

Comment:
I was thinking of something more like:

```java
verifyDirectives(Constants.PROVIDE_CAPABILITY,
   namespace -> "osgi.serviceloader".equals(namespace) ? "effective:|uses:|register:" : "effective:|uses:",
   null, null);
```
The namespace check is simple today. When we need more, we can convert to a switch statement

```java
verifyDirectives(Constants.PROVIDE_CAPABILITY,
	namespace -> {
		switch (namespace) {
			case "osgi.serviceloader" :
				return "effective:|uses:|register:";
			default :
				return "effective:|uses:";
		}
	},
	null, null);
```